c7n/resources/rds.py

```diff
@@ -281,6 +282,107 @@ def process(self, dbs):
                 **params)
 
 
+@filters.register('upgrade-available')
+class UpgradeAvailable(Filter):
+
+    schema = type_schema('upgrade-available')
+
+    def _list_versions(self, client, resource):
+        """ Scan DB instances for available engine upgrades
+
+            This will pull DB instances & check their specific engine for any
+            engine version with higher release numbers than the current one
+
+            This will also tag the rds instance with 'target_engine' which is
+            the most recent version of the engine available
+        """
+        version = resource['EngineVersion']
+        versions = client.describe_db_engine_versions(
+            Engine=resource['Engine'],
+            EngineVersion=version)['DBEngineVersions'][0]
```

Comment:
if we don't pass EngineVersion we can do one call per engine type. that might need some refactoring here which will invalidate some of the other review comments. we do want this accessible i think as  just a function that both this class and the upgrade class can call (if annotation not present).